### ISSUE TITLE
demo site for keras-cv layers 

### ISSUE THREAD

**Issue by pure-rgb** [keras-team/keras-cv#1801]
there are lots of preprocessing and augmentation layers that are hard to understand. can keras offer some like https://demo.albumentations.ai/ - an interactive site.

**Comment by jbischof** [comment]
Agree this would be cool, but we don't have the bandwidth unfortunately. Is this something you could contribute @pure-rgb?

**Comment by pure-rgb** [comment]
Yes, I would love to contribute. This should be live in keras-io.
cc. @fchollet

**Comment by jbischof** [comment]
@pure-rgb does this proposal require any changes to the codebase? What if you make a demo site for a handful of augmentations so I can get feedback from the team? Thank you!

**Comment by pure-rgb** [comment]
@IMvision12 I'm not free at the moment, but hopefully I will catch up ASAP.  If you want you can start working on it, and later we can collaborate for sure.

**Comment by IMvision12** [comment]
Hey @pure-rgb thanks I have started, and almost completed with frontend and backend just solving some errors

**Comment by james77777778** [comment]
Hey @IMvision12 @pure-rgb @jbischof 
I would like to share some code that could be helpful in creating a demo site using Streamlit.
https://github.com/james77777778/keras-aug/tree/main/streamlit_app
- `app.py` contains the main logic and layout of the site
- `layers_config.py` contains the configuration of the layers

Live demo on Streamlit cloud:
https://james77777778-keras-aug-streamlit-appapp-mxd7v1.streamlit.app/

As KerasAug is built on KerasCV, it should be easy to follow the implementation.

P.S. I'm willing to improve the preprocessing layers in KerasCV if the maintainers consider the implementation in KerasAug to be beneficial

**Comment by IMvision12** [comment]
@james77777778 Thank you for sharing this; I was planning to develop a website using HTML, CSS, JS, ReactJs, and Flask, but I believe using Streamlit will save time and be simpler.

**Comment by IMvision12** [comment]
@james77777778 BTW, do you intend to add bbox augmentation support to your website?

**Comment by james77777778** [comment]
> @james77777778 BTW, do you intend to add bbox augmentation support to your website?

Yes, it's part of my plan, but it might take some time.

**Comment by IMvision12** [comment]
https://github.com/IMvision12/kerascv-demo

did some changes, but major code was taken from https://github.com/james77777778/keras-aug/tree/main/streamlit_app
Thanks to @james77777778

@ianstenbit @jbischof

**Comment by jbischof** [comment]
This is super cool! I think the option to add a relevant box/mask to the image to see that transformed as well would definitely add to the utility.

**Comment by ianstenbit** [comment]
+1 -- this is awesome! Just played around with it a little bit. If this is easy to host we could maybe add it to KerasIO or something like that and link to it from our docs.

Great stuff 😄

**Comment by IMvision12** [comment]
@jbischof Yeah I am working on adding option for box and mask

**Comment by IMvision12** [comment]
Added bounding box support! currently, it supports "xywh" and "xyxy" formats!

![bbox](https://github.com/keras-team/keras-cv/assets/88665786/f8255a71-3922-46dd-b4d4-ebdfc0bf42e5)

**Comment by IMvision12** [comment]
Added segmentation support :) !
![Capture](https://github.com/keras-team/keras-cv/assets/88665786/c2abbe22-82ed-4658-a7f6-5572fe0678a1)

**Comment by IMvision12** [comment]
@jbischof @ianstenbit Let me Know where should i add this app, in KerasCV by creating a new folder named streamlit_app or Keras.io

**Comment by ianstenbit** [comment]
> @jbischof @ianstenbit Let me Know where should i add this app, in KerasCV by creating a new folder named streamlit_app or Keras.io

How much effort does it take to deploy this locally / host it?
If it's something we could serve statically, would GitHub pages work as an option? (I don't really know anything about streamlit or the details of how serving this works).

Otherwise we can also just include a link to this in our README / docs.

The mask support looks awesome 😄

**Comment by innat** [comment]
I think this should be added to the Keras-io website. This way it will reach and attract more users.

**Comment by jbischof** [comment]
@innat we don't know how to do this and are asking

**Comment by IMvision12** [comment]
@ianstenbit @innat @jbischof We cannot host the app on GitHub pages reason: https://discuss.streamlit.io/t/hosting-streamlit-on-github-pages/356/3.

I'm not familiar with the hosting details of keras.io, so I'm uncertain if it is possible to deploy this application on keras.io.

Suggestion: we can create a folder in kerascv itself and host, that's how I have hosted the app from my repo: https://github.com/IMvision12/kerascv-demo

Details of hosting streamlit app : https://blog.streamlit.io/host-your-streamlit-app-for-free/

**Comment by pure-rgb** [comment]
> I'm not familiar with the hosting details of keras.io, so I'm uncertain if it is possible to deploy this application on keras.io.

@fchollet

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new KerasHub package, which supports image, text, and multi-modal models. Please read https://github.com/keras-team/keras-hub/issues/1831. KerasHub will support all the core functionality of KerasCV.

KerasHub can be installed with !pip install -U keras-hub. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).

**Comment by innat** [comment]
@sachinprasadhs 
> If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).

Firstly, identifying which feature is required or missing can be done effectively, either by practitioners or the Keras team. Tickets in `keras-cv` were created only when such features were **NOT** available or when a specific need was encountered. Commenting after a significant delay and requesting the recreation of the same issue is neither efficient nor appropriate. Kindly move this ticket (and any other similar tickets with the same context) from `keras-cv` to `keras-hub`.
